edd-discount-widget.php

```diff
@@ -134,21 +141,69 @@ function form( $instance ) {
         }
 
 
+        /**
+         * Obtains the EDD discounts from the site URL.
+         * 
+         * @param mixed $site_url      The URL to extract the discounts from.
+         * @param mixed $api_key       The API key for the site.
+         * @param mixed $api_token     The API token for the site.
+         * @param int   $max_discounts The maximum number of discounts to obtain.
+         * 
+         * @return array|WP_Error      An array of discounts or a WP_Error object.
+         */
         function get_discounts( $site_url, $api_key, $api_token, $max_discounts ) {
+
+            $site_url      = sanitize_text_field( $site_url );
+            $api_key       = sanitize_text_field( $api_key );
+            $api_token     = sanitize_text_field( $api_token );
+            $max_discounts = absint( $max_discounts );
+
+            if ( empty( $site_url ) || empty( $api_key ) || empty( $api_token ) ) {
+                return new WP_Error(
+                    'edd-discounts-no-settings',
+                    __( 'Widget settings are incomplete.', 'edd-discounts-widget' )
+                );
+            }
+
             $options = array(
                 'timeout'   => 5
             );
 
             $temp_url = parse_url( $site_url );
-            if( !$temp_url['scheme'] == 'http' && !$temp_url['scheme'] == 'https' )
-                $site_url = 'http://' . $site_url;
+            if ( isset( $temp_url['scheme'] ) ) {
+                if( !$temp_url['scheme'] == 'http' && !$temp_url['scheme'] == 'https' ) {
+                    $site_url = 'http://' . $site_url;
+                }
+            }
 
             $discounts = wp_remote_get( $site_url . '/edd-api/discounts?key=' . rawurlencode( $api_key ) . '&token=' . rawurlencode( $api_token ) . '&format=json', $options );
-            $discounts = $discounts['body'];
 
-            if( !$discounts || is_wp_error( $discounts ) ) $discounts = __( 'An unknown error occurred!', 'edd-discounts-widget' );
+            if ( is_wp_error( $discounts ) ) {
+                return $discounts;
+            }
+
+            $response_body = json_decode( wp_remote_retrieve_body( $discounts ), true );
 
-            return $discounts;
+            if ( is_wp_error( $response_body ) ) {
```

Comment:
I don't think `$response_body` will ever be a `WP_Error` at this point. (It can be directly from `wp_remote_get()` as you've handled above, but not once you run `json_decode()` on it.